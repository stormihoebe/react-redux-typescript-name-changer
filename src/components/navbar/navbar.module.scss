@import "../../assets/colors.scss";

// navbar colors
$nav-bar-app-selector-color: $white;

.max_width {
    max-width: 1260px;
}

.nav_bar {
    width: 100%;
    display: flex;
    justify-content: center;
    background-color: $nav-bar-bg;
    color: $nav-bar-color;
}

.container {
    display: flex;
    flex: 1 0 100%;
    justify-content: space-between;
    composes: max_width;
}

.app_selector_icon {
    display: flex;
    align-items: center;
    margin-right: 11px;
}

.app_selector {
    display: flex;
    flex-direction: column;
    color: $nav-bar-app-selector-color !important;
    margin-right: 12px;
}

.app_selector_button {
    display: flex;
    align-items: center;
    height: 100%;
    margin-left: -10px;

    span {
        font-family: "Nunito Sans";
        line-height: 1;
        text-align: left;
    }

    &:focus {
        outline: none;
    }
}

.app_menu_popover {
    height: 100%;
}

.app_selector_title {
    font-size: 13px;
}

.app_selector_subtitle {
    font-size: 11px;
}

.app_selector_menu {
    min-width: 360px;

    li {
        display: flex;
        flex: 1 0 100%;
        justify-content: space-between;
    }

    .menu_item {
        display: flex;
        align-items: center;
        justify-content: space-between;
        width: 100%;

        img {
            margin-right: 9px;
        }

        .item_left {
            display: flex;
            align-items: center;
        }
    }
}

.user_menu {
    color: $user-menu-color !important;
    padding-right: 0;
}

.user_menu_chevron {
    color: $user-menu-chev-color !important;
}

.user_name {
    padding-right: 12px;
}