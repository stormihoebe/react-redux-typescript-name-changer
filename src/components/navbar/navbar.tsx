import * as React from "react"

const styles = require("./navbar.module.css")

interface NavbarProps {
    readonly logout: () => void,
    readonly userName: string,
}

export function Navbar(props: NavbarProps): JSX.Element {
    const { userName, logout } = props

    return(
        <nav className={`pt-navbar .modifier ${styles.nav_bar}`}>
            <div className={styles.container}>
                <div className="pt-navbar-group pt-align-right">
                    <button className={`pt-button pt-minimal ${styles.user_menu}`}>
                        <span className={styles.user_name} onClick={logout}>{userName}</span>
                        <span
                            className={`pt-icon-standard pt-icon-chevron-down ${styles.user_menu_chevron}`}
                        />
                    </button>
                </div>
            </div>
        </nav>
    )
}