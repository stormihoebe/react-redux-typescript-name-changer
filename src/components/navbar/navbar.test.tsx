import * as React from "react"
import * as ReactDOM from "react-dom"
import { Navbar } from "./navbar"
import { create as renderer } from "react-test-renderer"

const logout = () => {
    // does nothing
}

const renderNavbar = () => (
     <Navbar
        userName="Test"
        logout={logout}
    />
)

it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(renderNavbar(), div)
})

it("renders a snapshot", () => {
    const tree = renderer(renderNavbar()).toJSON()
    expect(tree).toMatchSnapshot()
})
