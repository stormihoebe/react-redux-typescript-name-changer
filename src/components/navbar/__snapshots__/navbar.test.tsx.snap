exports[`test renders a snapshot 1`] = `
<nav
  className="pt-navbar .modifier nav_bar">
  <div
    className="container">
    <div
      className="pt-navbar-group pt-align-right">
      <button
        className="pt-button pt-minimal user_menu">
        <span
          className="user_name"
          onClick={[Function]}>
          Test
        </span>
        <span
          className="pt-icon-standard pt-icon-chevron-down user_menu_chevron" />
      </button>
    </div>
  </div>
</nav>
`;
